test/e2e/pkg/grammar/clean-start/lexer/lexer.go

```diff
@@ -0,0 +1,1141 @@
+
```

Comment:
The `lexer`, `parser`, `sppf`, and `token` libraries all contain _only_ auto-generated code, which should be opaque to the user. Is this true and accurate?

If yes, could we:
1) Create a directory `grammar-auto` at this level and move all four dirs inside? (the intention is to show a user navigating files that they shouldn't inspect inside as they'll only find auto-generated code.
2) Add a small README.md at this level, explaining the only methods that are relevant outside this directory (the ones you use in `checker.go`

I'm really not sure how to go about 1). Other options would be:

* just move `abci_grammar_clean_start.md` up one level, and rename directory `clean-start` to `abci_grammar_clean_start`
* or the other way around: move `abci_grammar_clean_start.md` up one level and rename it to `clean-start`

The point is for the directory structure to give the user a hint of "stop opening subfolders, this is all generated code"